jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -77,19 +85,23 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
 
         path = pathlib.Path(data)
 
-        if path.is_file():
+        if path.is_dir() or load_as_list:
+            data = SpectrumList.read(str(path), format=format)
+            if data == []:
+                raise ValueError(f"`specutils.SpectrumList.read('{str(path)}')` "
+                                 "returned an empty list")
+        elif path.is_file():
             try:
-                data = [Spectrum1D.read(str(path), format=format)]
-                data_label = [app.return_data_label(data_label, alt_name="specviz_data")]
+                data = Spectrum1D.read(str(path), format=format)
+                if data.flux.ndim == 2:
+                    data, data_label = split_spectrum_with_2D_flux_array(data, data_label, app)
```

Comment:
In our file case, there are 2 data extensions, with the 1st having a 1d flux array, and the 2nd one having the nd flux array.  Without the `load_as_list` flag, this block only loads the first extension and never reaches line 97.  With `load_as_list` set, I get the error 
```
File [~/Work/jdaviz/jdaviz/configs/specviz/plugins/parsers.py:129](http://localhost:8888/lab/workspaces/auto-b/tree/~/Work/jdaviz/jdaviz/configs/specviz/plugins/parsers.py#line=128), in specviz_spectrum1d_parser(app, data, data_label, format, show_in_viewer, concat_by_file, cache, local_path, timeout, load_as_list)
    127 for wlind in range(len(spec.spectral_axis)):
    128     wlallorig.append(spec.spectral_axis[wlind].value)
--> 129     fnuallorig.append(spec.flux[wlind].value)
    131     # because some spec in the list might have uncertainties and
    132     # others may not, if uncert is None, set to list of NaN. later,
    133     # if the concatenated list of uncertanties is all nan (meaning
    134     # they were all nan to begin with, or all None), it will be set
    135     # to None on the final Spectrum1D
    136     if spec.uncertainty[wlind] is not None:

File [~/anaconda3/envs/newsv/lib/python3.11/site-packages/astropy/units/quantity.py:1302](http://localhost:8888/lab/workspaces/auto-b/tree/~/anaconda3/envs/newsv/lib/python3.11/site-packages/astropy/units/quantity.py#line=1301), in Quantity.__getitem__(self, key)
   1297     return self._new_view(
   1298         self.view(np.ndarray)[key], self.unit[key], propagate_info=False
   1299     )
   1301 try:
-> 1302     out = super().__getitem__(key)
   1303 except IndexError:
   1304     # We want zero-dimensional Quantity objects to behave like scalars,
   1305     # so they should raise a TypeError rather than an IndexError.
   1306     if self.isscalar:

IndexError: index 1 is out of bounds for axis 0 with size 1
```

Reply:
I think what I've done should fix this bug. It occurs when there's 2D spectra in a `SpectrumList`.

Reply:
Yes, with your PR it loads correctly, but your PR is basically unpacking the nd arrays into individual Spectrum1D objects, which this draft code is also doing.  In principle, with the change here, we don't need to manually unpack the nd fluxes inside the loader.  

This draft code should work loading nd-flux data more generally, outside of the specific mwm loader

Reply:
Yes, I've already reverted that back to how it was initially, see [specutils#1185, ca4d8c](https://github.com/astropy/specutils/pull/1185/commits/ca4d8c3394f189a8197c225f131b6efa6c5a8570)